src/Microsoft.ML.Data/Data/Conversion.cs

```diff
@@ -1243,7 +1241,12 @@ public bool TryParse(ref TX src, out I1 dst)
         {
             dst = default;
             TryParseSigned(I1.MaxValue, ref src, out long? res);
-            Contracts.Check(res.HasValue, "Value could not be parsed from text to sbyte.");
+            if (res == null)
+            {
+                dst = default;
+                return false;
+            }
+            Contracts.Assert(res.HasValue);
             Contracts.Check((I1)res == res, "Overflow or underflow occured while converting value in text to sbyte.");
             dst = (I1)res;
             return true;
```

Comment:
Is Contracts.Check sufficient enough for the cast? Contract.Check just checks if the statement is false and throws an exception. This will first try to convert a res to I1 before it does a conditional check. Therefore if the casting fails, could an invalid cast exception be thrown? If so, all of these should catch that and return false when caught.

Reply:
Disregard - TryParseSigned will set result to null if the parse value is about the max value.

---
In reply to: [228271251](https://github.com/dotnet/machinelearning/pull/1375#discussion_r228271251) [](ancestors = 228271251)